components/service/glean/docs/metrics/testing-metrics.md

```diff
@@ -0,0 +1,92 @@
+# Unit testing glean metrics
+
+In order to support unit testing inside of client applications using glean, a set of testing API 
+functions have been included.  The intent is to make glean easier to test 'out of the box' in any
+client application it may be used in.  These functions expose a way to inspect and validate recorded
+metric values within the client application but are restricted to test code only through visibility
+annotations (`@VisibleForTesting(otherwise = VisibleForTesting.NONE)`).
+
+## General test API method semantics
+
+In order to prevent issues with async calls when unit testing glean, it is important to put glean
+into testing mode by calling `Glean.enableTestingMode()`.  This will ensure that metrics are done
+recording when the other test functions are used.
+
+To check if a value exists (i.e. it has been recorded), there is a `testHasValue()` function on each 
+of the metric instances.
+
+Example:
```

Comment:
Last nit, i promise!
How about removing all the "Example:" lines? You can save some whitespace etc. by using:
> To check if a value exists (i.e. it has been recorded), there is a `testHasValue()` function on each 
of the metric instances:
> ```kotlin
> assertTrue(GleanMetrics.Foo.UriCount.testHasValue())
> ```